server/prometheus.js

```diff
@@ -50,14 +26,108 @@ class Prometheus {
         };
     }
 
+    /**
+     * Initialize Prometheus metrics, and add all available tags as possible labels.
+     * This should be called once at the start of the application.
+     * New tags will NOT be added dynamically, a restart is sadly required to add new tags to the metrics.
+     * Existing tags added to monitors will be updated automatically.
+     * @returns {Promise<void>}
+     */
+    static async init() {
+        // Add all available tags as possible labels,
+        // and use Set to remove possible duplicates (for when multiple tags contain non-ascii characters, and thus are sanitized to the same label)
+        const tags = new Set((await R.findAll("tag")).map((tag) => {
+            return Prometheus.sanitizeForPrometheus(tag.name);
+        }).filter((tagName) => {
+            return tagName !== "";
+        }));
+
+        const commonLabels = [
+            ...tags,
+            "monitor_id",
+            "monitor_name",
+            "monitor_type",
+            "monitor_url",
+            "monitor_hostname",
+            "monitor_port",
+        ];
```

Comment:
The tags here are now unique (good catch), but not sorted.
Please sort them to test that it works correctly